router/router.go

```diff
@@ -139,6 +139,28 @@ type Advert struct {
 	Events []*Event
 }
 
+// Strategy is route advertisement strategy
+type Strategy int
+
+const (
+	// All advertises all routes to the network
+	All Strategy = iota
```

Comment:
Might want to call these AdvertiseAll/AdvertiseOptimal so when they're used they look like:

```
router.AdvertiseAll
router.AdvertiseOptimal
```